### DISCUSSION TITLE
3090 fe vs 5090 fe training

### DISCUSSION THREAD

**Discussion by breezerc** [ultralytics/ultralytics#21515] [Q&A] [answered]
I got 5090 FE and compared training vs 3090 FE. Please find results below:
3090 training:
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/100      3.83G      1.126      1.357      1.171        163        640: 100%|██████████| 7393/7393 [18:16<00:00,  6.74it/s]  
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 157/157 [00:34<00:00,  4.57it/s]
                   all       5000      36335      0.569      0.424      0.449      0.309

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/100      4.15G      1.249      1.667      1.253        133        640: 100%|██████████| 7393/7393 [17:13<00:00,  7.15it/s]  
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 157/157 [00:30<00:00,  5.11it/s]
                   all       5000      36335      0.564      0.402      0.428      0.293

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/100      4.17G      1.343      1.914      1.321        247        640: 100%|██████████| 7393/7393 [16:35<00:00,  7.43it/s]  
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 157/157 [00:30<00:00,  5.14it/s]
                   all       5000      36335      0.528      0.381        0.4      0.272

5090:
Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/100      3.27G      1.126      1.356      1.172        163        640: 100%|██████████| 7393/7393 [17:46
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 157/157 [00:31<00:00,  5.03it/s]
                   all       5000      36335      0.569      0.434      0.455      0.313

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/100      3.55G       1.25      1.667      1.254        133        640: 100%|██████████| 7393/7393 [16:57<00:00,  7.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 157/157 [00:27<00:00,  5.72it/s]
                   all       5000      36335       0.56      0.407      0.428      0.293
                   
I'm confused that there is no big difference. It is exactly the same machine I just swapped cards. Running under WSL2/Ubuntu 24.04 with Windows 11 as host.

With 3090 I installed ultralytics/yolo by simple pip ultralytics
With 5090 I did this:
--extra-index-url https://download.pytorch.org/whl/cu128
torch
torchvision
torchaudio
ultralytics

Any ideas?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @breezerc, thank you for sharing your detailed comparison and for your interest in Ultralytics 🚀! We appreciate your initiative in benchmarking the 3090 FE and 5090 FE GPUs for training.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If this is a custom training ❓ Question, please include as much context as possible, such as your dataset structure, training configuration, relevant logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For new users or additional usage examples, check out the [Docs](https://docs.ultralytics.com/) for comprehensive [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) guides.

Join the Ultralytics community on [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat, visit [Discourse](https://community.ultralytics.com/) for in-depth discussions, or check out our [Subreddit](https://www.reddit.com/r/Ultralytics/) for knowledge sharing.

## Upgrade

To ensure you are using the latest features and fixes, please upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help you get started. An Ultralytics engineer will review your discussion and provide further assistance soon.

**Reply by breezerc** [reply]
Here is how I run the training:
from ultralytics import YOLO

# Load a model
model = YOLO("yolo11n.pt")  # load a pretrained model (recommended for training)

# Train the model
results = model.train(data="coco.yaml", epochs=100, imgsz=640)

I wiped out venv and then recreated and installed pytorch and ultralytics to be able to run it using 5090.

**Reply by glenn-jocher** [reply, marked as answer]
@breezerc the similar performance you're seeing is actually expected for YOLO11n with these settings. The RTX 5090's additional compute power isn't being fully utilized because you're likely hitting other bottlenecks like data loading, CPU preprocessing, or the model size itself.

To better leverage the 5090's capabilities, try increasing your batch size with `batch=32` or higher (the 5090 has 32GB VRAM vs 3090's 24GB), use a larger model like `yolo11m.pt` or `yolo11l.pt`, or enable mixed precision training with `amp=True`. You can also try larger image sizes like `imgsz=1280` to stress the GPU more and see the performance difference.

**Reply by breezerc** [reply]
@glenn-jocher got it. thanks!

**Reply by glenn-jocher** [reply]
@breezerc you're welcome! Happy training with your new 5090 🚀
